assessment/collectors/parser/response_parser_test.go

```diff
@@ -0,0 +1,984 @@
+/* Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.*/
+
+package assessment
+
+import (
+	"reflect"
+	"testing"
+
+	. "github.com/GoogleCloudPlatform/spanner-migration-tool/assessment/utils"
+	"github.com/GoogleCloudPlatform/spanner-migration-tool/logger"
+	"go.uber.org/zap"
+)
+
+func init() {
+	logger.Log = zap.NewNop() // Set to a no-op logger during tests
+}
+func TestParseStringArrayInterface(t *testing.T) {
+	tests := []struct {
+		name  string
+		input any
+		want  []string
+	}{
+		{
+			name:  "string array",
+			input: []string{"a", "b", "c"},
+			want:  []string{"a", "b", "c"},
+		},
+		{
+			name:  "single string",
+			input: "abc",
+			want:  []string{"abc"},
+		},
+		{
+			name:  "interface array of strings",
+			input: []any{"a", "b", "c"},
+			want:  []string{"a", "b", "c"},
+		},
+		{
+			name:  "interface array of mixed types",
+			input: []any{"a", 1, "c", 2.0},
+			want:  []string{"a", "c"},
+		},
+		{
+			name:  "empty interface array",
+			input: []any{},
+			want:  []string{},
+		},
+		{
+			name:  "nil input",
+			input: nil,
+			want:  []string{},
+		},
+		{
+			name:  "interface array with nil",
+			input: []any{"a", nil, "c"},
+			want:  []string{"a", "c"},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got := ParseStringArrayInterface(tt.input)
+			if !reflect.DeepEqual(got, tt.want) {
+				t.Errorf("ParseStringArrayInterface() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func TestParseAnyToString(t *testing.T) {
+	tests := []struct {
+		name  string
+		input any
+		want  string
+	}{
+		{
+			name:  "string",
+			input: "abc",
+			want:  "abc",
+		},
+		{
+			name:  "integer",
+			input: 123,
+			want:  "123",
+		},
+		{
+			name:  "float",
+			input: 3.14,
+			want:  "3.14",
+		},
+		{
+			name:  "boolean",
+			input: true,
+			want:  "true",
+		},
+		{
+			name:  "nil",
+			input: nil,
+			want:  "<nil>",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got := parseAnyToString(tt.input)
+			if got != tt.want {
+				t.Errorf("parseAnyToString() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func TestParseAnyToInteger(t *testing.T) {
+	tests := []struct {
```

Comment:
nit - can you move this into a common struct with all the use cases ?

Reply:
All the test cases have a bit different struct and may be modified in future. We should define localized struct. 
Reference: https://engdoc.corp.google.com/eng/doc/tott/episodes/643.md?cl=head